core/trino-main/src/main/java/io/trino/sql/planner/plan/TableFunctionProcessorNode.java

```diff
@@ -70,6 +76,9 @@ public class TableFunctionProcessorNode
 
     private final TableFunctionHandle handle;
 
+    @Nullable // null on workers
+    private final TupleDomain<ColumnHandle> enforcedConstraint;
```

Comment:
i am not convinced `ColumnHandle` should be the TupleDomain key here.

- `TableFunctionAnalysis.returnedType` describes the relation in terms of Fields with optional names so they would be positionally discernable (`TupleDomain<Integer>` ...)
- `properOutputs` field captures the symbols produced by the function, and we could use that ( `TupleDomain<Symbol>`)

i think this should be `TupleDomain<Symbol>`

Reply:
It will be `TupleDomain<Integer>`